modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/GridCacheFullTextQuerySelfTest.java

```diff
@@ -212,86 +311,109 @@ private static void clearCache(IgniteEx ignite) {
+                TestPair testPair = processExpectedWithBinary(exp, cursor);
 
-                    assertEquals(entry.getKey(), entry.getValue().field("age"));
-
-                    exp0.remove(entry.getKey());
-                }
-
-                checkForMissedKeys(ignite, exp0, all);
+                assertResult(ignite, qry, testPair);
             }
 
             try (QueryCursor<Cache.Entry<Integer, BinaryObject>> cursor = cache0.query(qry)) {
-                Set<Integer> exp0 = new HashSet<>(exp);
-
-                List<Cache.Entry<Integer, ?>> all = new ArrayList<>();
 
-                for (Cache.Entry<Integer, BinaryObject> entry : cursor.getAll()) {
-                    all.add(entry);
+                TestPair testPair = processExpectedWithBinary(exp, cursor);
 
-                    assertEquals(entry.getKey().toString(), entry.getValue().field("name"));
-
-                    assertEquals(entry.getKey(), entry.getValue().field("age"));
-
-                    exp0.remove(entry.getKey());
-                }
-
-                checkForMissedKeys(ignite, exp0, all);
+                assertResult(ignite, qry, testPair);
             }
         }
         else {
             try (QueryCursor<Cache.Entry<Integer, Person>> cursor = cache.query(qry)) {
-                Set<Integer> exp0 = new HashSet<>(exp);
 
-                List<Cache.Entry<Integer, ?>> all = new ArrayList<>();
+                TestPair testPair = processExpected(exp, cursor);
 
-                for (Cache.Entry<Integer, Person> entry : cursor.getAll()) {
-                    all.add(entry);
+                assertResult(ignite, qry, testPair);
 
-                    assertEquals(entry.getKey().toString(), entry.getValue().name);
+            }
 
-                    assertEquals(entry.getKey(), Integer.valueOf(entry.getValue().age));
+            try (QueryCursor<Cache.Entry<Integer, Person>> cursor = cache.query(qry)) {
 
-                    exp0.remove(entry.getKey());
-                }
+                TestPair testPair = processExpected(exp, cursor);
 
-                checkForMissedKeys(ignite, exp0, all);
+                assertResult(ignite, qry, testPair);
             }
+        }
+    }
 
-            try (QueryCursor<Cache.Entry<Integer, Person>> cursor = cache.query(qry)) {
-                Set<Integer> exp0 = new HashSet<>(exp);
+    /**
+     * Checks query for missed keys or if limit is set - for limitation correctness.
+     *
+     * @param ignite Ignite context.
+     * @param qry Initial text query.
+     * @param testPair pair containing expected and all entries.
+     * @throws IgniteCheckedException if key check failed.
+     */
+    private static void assertResult(IgniteEx ignite, TextQuery qry,
+        TestPair testPair) throws IgniteCheckedException {
+        if (qry.getLimit() > 0) {
+            assertTrue(testPair.all.size() <= QUERY_LIMIT);
+        }
+        else {
```

Comment:
According to [code style](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-BracesandIdentation) braces should not be used for single-line statements after _if_ and _else_.